keylime/cloud_verifier_tornado.py

```diff
@@ -609,43 +595,58 @@ def post(self):
                         return
 
                     # Write IMA policy to database if needed
-                    if not ima_policy_name and not ima_policy:
-                        logger.info("Allowlist data not provided with request! Using default empty allowlist.")
-                        ima_policy = json.dumps(ima.process_ima_policy(ima.EMPTY_ALLOWLIST, []))
-
-                    if ima_policy:
-                        err_msg = cloud_verifier_common.validate_ima_policy_data(ima_policy)
-                        if err_msg:
-                            web_util.echo_json_response(self, 400, err_msg)
-                            logger.warning(err_msg)
+                    if not runtime_policy_name and not runtime_policy:
+                        logger.info("IMA policy data not provided with request! Using default empty IMA policy.")
+                        runtime_policy = json.dumps(ima.EMPTY_RUNTIME_POLICY)
+
+                    if runtime_policy:
+                        runtime_policy_sig = json_body.get("runtime_policy_sig")
+                        runtime_policy_key = json_body.get("runtime_policy_key")
+
+                        if runtime_policy_key and runtime_policy_sig:
+                            runtime_policy_key = base64.b64decode(runtime_policy_key)
+                            runtime_policy_sig = base64.b64decode(runtime_policy_sig)
+
```

Comment:
Had problems with typecheckers changing variable types... here from `str` to `bytes`.

Reply:
@stefanberger Would you mind clarifying what the issue is a little more? I'm not super familiar with typecheckers yet.

Is it that I need to create separate variables for the `bytes`-decoded key and sig, instead of reusing the existing `str` variables?

Reply:
Sorry, for some reason I didn't seem to get email notifications about your messages here. I would have reacted a bit faster.

What typecheckers don't seem to like is if variables are first of one type and then they change to another one. So, what you did looks good.